src/Compilers/RealParserTests/RandomRealParserTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```

Comment:
The search space may be countably infinite in theory, (because you can always add another decimal digit of precision), but in practice there's a limit on the length of strings and the memory used by BigInteger. I think that makes it countable (even if the maximum string length is 2^32).

:+1: on it being a good validation if we ever change the algorithm. During development, did those bugs pop out quickly, or did you have to run this code for hours or days? You should probably make a comment on that so that future maintainers aren't left guessing how long they should let it run to get to the same level of confidence.